web-services/query/src/main/java/datawave/webservice/query/logic/filtered/FilteredQueryLogic.java

```diff
@@ -0,0 +1,383 @@
+ * A filtered query logic will only actually execute the delegate query logic if the filter passes. Otherwise this will do nothing and return no results.
+ */
+public class FilteredQueryLogic implements QueryLogic<Object> {
+    
+    private QueryLogic<Object> delegate;
+    private QueryLogicFilter filter;
+    
+    private boolean filtered = false;
+    
+    public FilteredQueryLogic() {}
+    
+    public FilteredQueryLogic(FilteredQueryLogic other) throws CloneNotSupportedException {
+        this.delegate = (QueryLogic<Object>) (other.delegate.clone());
+        this.filter = other.filter;
+        this.filtered = other.filtered;
+    }
+    
+    public QueryLogic<Object> getDelegate() {
+        return delegate;
+    }
+    
+    public void setDelegate(BaseQueryLogic<Object> delegate) {
+        this.delegate = delegate;
+    }
+    
+    public QueryLogicFilter getFilter() {
+        return filter;
+    }
+    
+    public void setFilter(QueryLogicFilter filter) {
+        this.filter = filter;
+    }
+    
+    public interface QueryLogicFilter {
+        boolean canRunQuery(Query settings, Set<Authorizations> auths);
+    }
+    
+    @Override
+    public String getPlan(Connector connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations, boolean expandFields, boolean expandValues)
+                    throws Exception {
+        if (!filtered && filter.canRunQuery(settings, runtimeQueryAuthorizations)) {
+            return delegate.getPlan(connection, settings, runtimeQueryAuthorizations, expandFields, expandValues);
+        } else {
+            filtered = true;
+            return "";
+        }
+    }
+    
+    @Override
+    public GenericQueryConfiguration initialize(Connector connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations) throws Exception {
+        if (!filtered && filter.canRunQuery(settings, runtimeQueryAuthorizations)) {
+            return delegate.initialize(connection, settings, runtimeQueryAuthorizations);
+        } else {
+            filtered = true;
+            GenericQueryConfiguration config = new GenericQueryConfiguration() {};
+            config.setConnector(connection);
+            config.setQueryString("");
+            config.setAuthorizations(runtimeQueryAuthorizations);
+            config.setBeginDate(settings.getBeginDate());
+            config.setEndDate(settings.getEndDate());
+            return config;
+        }
+    }
+    
+    @Override
+    public void setupQuery(GenericQueryConfiguration configuration) throws Exception {
+        if (!filtered) {
+            delegate.setupQuery(configuration);
+        }
+    }
+    
+    @Override
+    public Iterator<Object> iterator() {
+        if (!filtered) {
+            return delegate.iterator();
+        } else {
+            return Collections.emptyIterator();
+        }
+    }
+    
+    public List<String> getSelectors(Query settings) {
+        return delegate.getSelectors(settings);
+    }
```

Comment:
Consider introducing a class called `DelegatingQueryLogic` or something that simply delegates to another query logic and would include most of the code in `FilteredQueryLogic`. Then `FilteredQueryLogic` can extend that and implement only those methods required for implementing filtering. This would simplify the implementation of `FilteredQueryLogic` and provide a clearer separation of concerns (at the expense of a more complex class hierarchy) - so it's may not be a new win to do this.

Reply:
Actually, I like that idea.  Doing it.